flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -1634,6 +1641,28 @@ public boolean updateState(TaskExecutionState state) {
 		}
 	}
 
+	private void maybeReleasePartitions(final Execution attempt, final TaskExecutionState state) {
+		final ExecutionVertexID finishedExecutionVertex = attempt.getVertex().getID();
+
+		if (state.getExecutionState() == ExecutionState.FINISHED) {
+			final List<IntermediateResultPartitionID> releasablePartitions = partitionReleaseStrategy.vertexFinished(finishedExecutionVertex);
+			releasePartitions(releasablePartitions);
+		} else {
+			partitionReleaseStrategy.vertexUnfinished(finishedExecutionVertex);
+		}
+	}
+
+	private void releasePartitions(final List<? extends IntermediateResultPartitionID> releasablePartitions) {
+		for (final IntermediateResultPartitionID resultPartition : releasablePartitions) {
+			releasePartitions(resultPartition);
+		}
+	}
+
+	private void releasePartitions(final IntermediateResultPartitionID resultPartition) {
```

Comment:
JM/TM track/release partitions by their `ResultPartitionID`. You likely will have to use that here as well.

Reply:
or let me ask it differently, how do you intend to map the `IntermediateResultPartitionID` to the `ResultPartitionID`?

Reply:
We will have to look up the `ExecutionVertex` of the `IntermediateResultPartitionID`, which could be done via the `SchedulingTopology`.